openpype/client/entities.py

```diff
@@ -518,6 +518,53 @@ def get_subset_families(project_name, subset_ids=None):
     return set()
 
 
+def match_subset_id(
```

Comment:
Not sure if this function should be here. Also it's name does not tell much about what it does? Does it find subset based on family, asset and task? If yes then it make sense only for very limited use cases. I guess it's used to find workfile subset in that case I would mark it as helper function in openpype/pipeline/workfiles.

BTW without Python 3 syntax. We still must support Python 2

Reply:
As I'm looking for the usage I would say the functionality should be moved to sync server module and `pre_copy_last_published_workfile.py` should be launch hook inside sync server too (with lower order so it's processed before the global one).

Reply:
Okay, currently there are no hook for `sync_server` right? Therefore you expect me to set the `launch_hooks` up with this one?

Reply:
Yes, sync server must implement `get_launch_hook_paths` to return the path to it. Also the functions related to workfile workflow in `openpype/client/entities` should be moved to sync module too. Those functions are utils functions for workfile sync workflow and it would be confusing to have them in public api.

BTW keep the original `pre_copy_last_published_workfile.py` without sync server logic in global hooks please.

Reply:
I would recommend renaming `match_subset_id` to `get_matching_subset_id` and `get_representation_by_task` should handle cases when representation does not have `"task"` key in `"context"`.

Reply:
>  Also the functions related to workfile workflow in openpype/client/entities should be moved to sync module too.

This is still true, both functions do not belong to `entities.py`. They are both dependent on specific workflows and expectations that are not requirements (e.g. task name in subset name or representation with task in context). As I'm thinking about it they are just filtering functions and they don't have to query from server at all `get_representation_by_task` could be changed to `fitler_representations_by_task` with `representations` as an argument and `get_matching_subset_id` also another filtering function `filter_subsets_by_family_and_task` with `subsets` as an argument.

Reply:
Okay, in fact this split into functions was made by anticipation for another feature #4176 but its implementation is still changing and uncertain, therefore we'll create these utils functions in this upcoming contribution if you don't mind, and keep this one light.

Reply:
Okay, I removed them from this PR, and we'll find a better place in the upcoming PR if we still even need these functions.
Thanks for your advises.